function-aws-api-proxy/src/main/java/io/micronaut/function/aws/proxy/MicronautLambdaPayloadV2Handler.java

```diff
@@ -0,0 +1,451 @@
+import io.micronaut.http.uri.UriBuilder;
+import io.micronaut.inject.qualifiers.Qualifiers;
+import io.micronaut.jackson.codec.JsonMediaTypeCodec;
+import io.micronaut.scheduling.executor.ExecutorSelector;
+import io.micronaut.web.router.RouteMatch;
+import io.micronaut.web.router.Router;
+import io.micronaut.web.router.UriRouteMatch;
+import io.micronaut.web.router.resource.StaticResourceResolver;
+import jdk.internal.joptsimple.internal.Strings;
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+@TypeHint(
+    accessType = {TypeHint.AccessType.ALL_DECLARED_CONSTRUCTORS, TypeHint.AccessType.ALL_PUBLIC},
+    value = {
+        APIGatewayV2HTTPEvent.class,
+        APIGatewayV2HTTPEvent.RequestContext.class,
+    }
+)
+public class MicronautLambdaPayloadV2Handler implements RequestHandler<APIGatewayV2HTTPEvent,
+    APIGatewayV2HTTPResponse>, ApplicationContextProvider {
+    private static final Logger LOG = LoggerFactory.getLogger(MicronautLambdaPayloadV2Handler.class);
+    private static final String TIMER_INIT = "MICRONAUT_COLD_START";
+    private static final String TIMER_REQUEST = "MICRONAUT_HANDLE_REQUEST";
+    private final ApplicationContextBuilder applicationContextBuilder;
+
+    private ApplicationContext applicationContext;
+
+    private LambdaContainerState lambdaContainerEnvironment;
+    private RequestArgumentSatisfier requestArgumentSatisfier;
+    private StaticResourceResolver resourceResolver;
+    private Router router;
+    private ErrorResponseProcessor errorResponseProcessor;
+    private RouteExecutor routeExecutor;
+    private HttpRequestFactory requestFactory;
+
+    public MicronautLambdaPayloadV2Handler() throws ContainerInitializationException {
+        this(ApplicationContext.builder());
+    }
+
+    public MicronautLambdaPayloadV2Handler(ApplicationContextBuilder applicationContextBuilder) throws ContainerInitializationException {
+        this.applicationContextBuilder = applicationContextBuilder;
+        this.lambdaContainerEnvironment = new LambdaContainerState();
+        initialize();
+    }
+
+    @Override
+    public APIGatewayV2HTTPResponse handleRequest(final APIGatewayV2HTTPEvent event,
+                                                  final Context context) {
+        HttpRequest<?> request = toHttpRequest(event);
+
+        Timer.start(TIMER_REQUEST);
+        HandlerUtils.configureWithContext(this, context);
+
+        try {
+            HttpResponse<?> response = ServerRequestContext.with(request, (Supplier<HttpResponse<?>>) () -> {
+                Optional<UriRouteMatch> routeMatch = request
+                    .getAttribute(HttpAttributes.ROUTE_MATCH, UriRouteMatch.class);
+
+                if (routeMatch.isPresent()) {
+                    return handleRouteMatch(routeMatch.get(), request);
```

Comment:
Is there a way to have Micronaut Environment as a function? Similar to `HTTPHandlerFunc` in Go?